### PR TITLE
usbus: Unify terminology to use 'descriptor' everywhere

### PR DESCRIPTION
**Pull request by bergzand**
### Contribution description

This PR unifies the used terminology within USBUS to match the terminology from the specs.
`hdr`, `hdrs`, `headers`, etc is replaced everywhere with `descr` `descriptors` and `descriptors`.
I've also renamed some formatter functions to make it more consistent and hopefully a bit more clear.

Two very minimal once used functions have been merged with their calling functions. Let me know if it is too much to piggy-back here.

*note*: As some enums and struct members have been modified, this is a small API change.

### Testing procedure

Most of the PR is renaming of structs, documentation and function names. Impact should be minimal.

For the review, please verify that 'descriptor' and related words are used everywhere, also please look at some of the new enum names and function names and check if they make sense from a somewhat more outsider point of view. Mainly if the members in `usbus_descr_gen_funcs_t` have  clear names for the reviewer. Suggestions welcome.

Testing if the descriptors generated before and after this PR are identical should be enough for functional testing.

### Issues/PRs references

~~depends on #12429 ~~

### CODE REVIEW

File: sys/include/usb/usbus.h [hunk 1]
```diff
@@ -232,7 +254,7 @@ typedef struct usbus_endpoint {
     struct usbus_endpoint *next;    /**< Next endpoint in the
                                          @ref usbus_interface_t list of
                                          endpoints */
-    usbus_hdr_gen_t *hdr_gen;       /**< Optional additional header generator */
+    usbus_descr_gen_t *descr_gen;     /**< Optional additional descriptor generator */
```
Comment (chrysn) [inline sys/include/usb/usbus.h]: Indentation level is broken here and in the equivalent lines below.

**Comment by bergzand** [comment]
> Regarding the current merge conflict in cdc_acm.c: This seems to be harmless in that where `fmt_pre_descriptor` is added, the new terminology is already used.

Resolved with a rebase.

File: sys/include/usb/usbus.h (hunk 1 above)
Comment (bergzand) [reply sys/include/usb/usbus.h]: All fixed, thanks

File: sys/include/usb/usbus.h [hunk 2]
```diff
@@ -281,7 +281,7 @@ typedef struct usbus_interface_alt {
 typedef struct usbus_interface {
     struct usbus_interface *next;       /**< Next interface (set by USBUS during
                                              registration)                     */
-    usbus_hdr_gen_t *hdr_gen;           /**< Optional additional header
+    usbus_descr_gen_t *descr_gen;       /**< Optional additional descriptor
                                              generators                        */
```
Comment (dylad) [inline sys/include/usb/usbus.h]: generator or generators ? Must be align with line 272 of the same file.

Comment (bergzand) [reply sys/include/usb/usbus.h]: Something something linked list of generators, clarified and fixed all occurences.

**Comment by bergzand** [comment]
>  Yet another rebase awaits me for MSC :)

Sorry! :sob:

**Comment by chrysn** [comment]
I'm happy with those changes, please squash (provided my comment on that suffices).

**Comment by bergzand** [comment]
squashed!

**Comment by dylad** [comment]
Please fix doxygen and codespell

**Comment by bergzand** [comment]
I've amended the commit with both the doxygen and the codespell issue fixed

**Comment by bergzand** [comment]
I clearly forgot to adapt parts of the cdc_acm code (or I mixed up a squash somewhere). Will fix tomorrow.

**Comment by dylad** [comment]
Please squash.

**Comment by bergzand** [comment]
Squashed!

**Pull request merged**